MapboxNavigation/RouteMapViewController.swift

```diff
@@ -273,131 +280,100 @@ extension RouteMapViewController: NavigationMapViewDelegate {
         hasFinishedLoadingMap = true
     }
 
-    @objc(navigationMapView:shouldUpdateTo:)
     func navigationMapView(_ mapView: NavigationMapView, shouldUpdateTo location: CLLocation) -> CLLocation? {
-
-        guard routeController.userIsOnRoute(location) else { return nil }
-        guard let stepCoordinates = routeController.routeProgress.currentLegProgress.currentStep.coordinates else  { return nil }
-        guard let snappedCoordinate = closestCoordinate(on: stepCoordinates, to: location.coordinate) else { return location }
-
-        // Add current way name to UI
-        if let style = mapView.style, recenterButton.isHidden && hasFinishedLoadingMap {
-            let closestCoordinate = snappedCoordinate.coordinate
-            let roadLabelLayerIdentifier = "roadLabelLayer"
-            var streetsSources = style.sources.flatMap {
-                $0 as? MGLVectorSource
-                }.filter {
-                    $0.isMapboxStreets
-            }
-            
-            // Add Mapbox Streets if the map does not already have it
-            if streetsSources.isEmpty {
-                let source = MGLVectorSource(identifier: "mapboxStreetsv7", configurationURL: URL(string: "mapbox://mapbox.mapbox-streets-v7")!)
-                style.addSource(source)
-                streetsSources.append(source)
-            }
+        guard let snappedLocation = routeController.snappedLocation else {
```

Comment:
I'd say the other way around, the route controller should drive the map view, but that might be out of scope for this PR?